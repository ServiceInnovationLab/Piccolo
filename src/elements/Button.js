import styled from 'styled-components';

const Button = styled.button`
  background: #019dff;
  color: #fff;
  font-weight: bold;
  text-align: center;
  border: 1px solid #019dff;
  border-radius: 50px;
  height: 74px;
  width: 100%;
  font-size: 34px;
  @media (min-width: 1024px) {
    width: 360px;
  }
`;

export default Button;
