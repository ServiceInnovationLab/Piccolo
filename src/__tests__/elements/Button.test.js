import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../elements/Button';
import 'jest-styled-components';

test('Button renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('background', '#019dff');
});
