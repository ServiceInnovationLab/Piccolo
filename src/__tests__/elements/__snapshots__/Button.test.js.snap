// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  background: #019dff;
  color: #fff;
  font-weight: bold;
  text-align: center;
  border: 1px solid #019dff;
  border-radius: 50px;
  height: 74px;
  width: 100%;
  font-size: 34px;
}

@media (min-width:1024px) {
  .c0 {
    width: 360px;
  }
}

<button
  className="c0"
/>
`;
